doc/source/whatsnew/v3.0.0.rst

```diff
@@ -641,6 +641,7 @@ Categorical
 
 Datetimelike
 ^^^^^^^^^^^^
+- Bug in :attr:`Series.dt.date` where Series with all NaT values would raise an error when compared to a datetime.date (:issue:`61188`)
```

Comment:
The root cause, and only change in behavior for pandas, is not returning object dtype. I think we should highlight that instead.

```suggestion
- Bug in :attr:`Series.dt.date` returned datetime dtype on Series with all NaT values; now returns object dtype (:issue:`61188`)
```